server/src/main/scala/de/frosner/broccoli/services/InstanceService.scala

```diff
@@ -306,4 +308,37 @@ class InstanceService @Inject()(templateService: TemplateService,
     tryDelete.map(addStatuses)
   }
 
+  /**
+    * Get all tasks of the given instance.
+    *
+    * In Nomad the hierarchy is normally "allocation -> tasks in that allocation".  However allocations have generic
+    * UUID whereas tasks have human-readable names, so we believe that tasks are easier as an "entry point" for the user
+    * in the UI.  Hence this method inverts the hierarchy of models returned by Nomad.
+    *
+    * @param id The instance ID
+    * @return All tasks of the given instance with their allocations, or an empty list if the instance has no tasks or
+    *         didn't exist.  If the user may not access the instance return an InstanceError instead.
+    */
+  def getInstanceTasks(user: Account)(id: String): EitherT[Future, InstanceError, InstanceTasks] =
+    EitherT
+      .pure[Future, InstanceError](id)
+      .ensureOr(InstanceError.UserRegexDenied(_, user.instanceRegex))(_.matches(user.instanceRegex))
+      .semiflatMap(nomadClient.getAllocationsForJob)
+      .map { allocations =>
+        InstanceTasks(
+          id,
+          // Invert the order "allocation -> task" into "task -> allocation" (see doc comment)
+          allocations.payload
+            .flatMap(allocation => allocation.taskStates.mapValues(_ -> allocation))
+            .groupBy(_._1)
```

Comment:
Well, that's just the name for the "first element of a pair"… what would you like me to use instead?